chart/kubeapps/templates/kubeappsapis/rbac.yaml

```diff
@@ -26,6 +27,11 @@ rules:
   - apiGroups: [""]
     resources: ["secrets"]
     verbs: ["get", "list"]
+  # needed by fluxv2 plug-in to check whether flux CRDs have been installed
+  # TODO (gfichtenholt) possibly surround this with "if fluxv2 plugin is enabled"
```

Comment:
Yes, but not as a condition on this ClusterRole... this cluster role is currently intended to be used in development only (ie. the binding below is only done when `unsafeUseDemoSA` is set to true), hence the name being `kubeapps-apis-dev` above. The helm (direct) and kapp-controller plugins are able to use the users' credential for all interactions when requests come from the client.

But in your case, because you're caching data and interacting with the cluster outside of user requests, you need to ensure that extra RBAC required by your plugin is available to the service account. So longer term (not for this PR), I'd separate it out into a different cluster-role (perhaps in a different file) and put the condition there.

Reply:
+1